### ISSUE TITLE
Error: SSR requires `@loadable/babel-plugin`, please install it

### ISSUE THREAD

**Issue by hazem3500** [graysonhicks/gatsby-plugin-loadable-components-ssr#31]
When I run `gatsby build` I get the error 
```bash
WebpackError: Invariant Violation: loadable: SSR requires `@loadable/babel-plugin`, please install it
```
I'm on version `3.2.0` and the gatsby version is `3.7.1`.

The loadable component is 
```jsx
import Loadable from "@loadable/component"

const DistortedShader = Loadable(() => import("./DistortedShader"))

export {DistortedShader}
```

And then I'm importing it inside an `MDX` file

```jsx
import { DistortedShader } from "../../../demos"

<DistortedShader />
```

Not sure what's the problem though 😅

**Comment by graysonhicks** [comment]
Hi @hazem3500 !  Can you post your `package.json` and `gatsby-config`?

**Comment by hazem3500** [comment]
```json
// package.json
  "dependencies": {
    "@chakra-ui/react": "^1.6.3",
    "@chakra-ui/theme-tools": "^1.1.7",
    "@emotion/react": "^11.4.0",
    "@emotion/styled": "^11.3.0",
    "@loadable/component": "^5.15.0",
    "@mdx-js/mdx": "^1.6.22",
    "@mdx-js/react": "^1.6.22",
    "@react-hook/window-scroll": "^1.3.0",
    "framer-motion": "4.1.17",
    "gatsby": "^3.7.1",
    "gatsby-image": "^3.7.1",
    "gatsby-plugin-advanced-sitemap": "^2.0.0",
    "gatsby-plugin-canonical-urls": "^3.7.1",
    "gatsby-plugin-feed": "^3.7.1",
    "gatsby-plugin-image": "^1.7.1",
    "gatsby-plugin-loadable-components-ssr": "^3.2.0",
    "gatsby-plugin-manifest": "^3.7.1",
    "gatsby-plugin-mdx": "^2.7.1",
    "gatsby-plugin-offline": "^4.7.1",
    "gatsby-plugin-plausible": "^0.0.7",
    "gatsby-plugin-react-helmet": "^4.7.1",
    "gatsby-plugin-react-svg": "^3.0.1",
    "gatsby-plugin-remove-trailing-slashes": "^3.10.0",
    "gatsby-plugin-robots-txt": "^1.6.2",
    "gatsby-plugin-sharp": "^3.7.1",
    "gatsby-plugin-twitter": "^3.7.1",
    "gatsby-remark-audio": "^1.0.2",
    "gatsby-remark-autolink-headers": "^4.4.1",
    "gatsby-remark-copy-linked-files": "^4.4.1",
    "gatsby-remark-embedder": "^5.0.0",
    "gatsby-remark-external-links": "^0.0.4",
    "gatsby-remark-images": "^5.4.1",
    "gatsby-remark-smartypants": "^4.4.1",
    "gatsby-remark-vscode": "^3.2.1",
    "gatsby-source-filesystem": "^3.7.1",
    "gatsby-transformer-remark": "^4.4.1",
    "gatsby-transformer-sharp": "^3.7.1",
    "gl-react": "^5.0.0",
    "gl-react-dom": "^5.1.0",
    "prop-types": "^15.7.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-helmet": "^6.1.0",
    "react-hook-form": "^7.8.3",
    "react-icons": "^4.2.0",
    "react-live": "^2.2.3",
    "react-query": "^3.16.1",
    "react-resize-detector": "^6.7.6",
    "react-use-gesture": "^9.1.3",
    "typeface-work-sans": "1.1.13",
    "use-animation-frame": "^0.1.0"
  },
  "devDependencies": {
    "@babel/core": "^7.14.5",
    "@loadable/babel-plugin": "^5.13.2",
    "@storybook/addon-actions": "^6.2.9",
    "@storybook/addon-essentials": "^6.2.9",
    "@storybook/addon-links": "^6.2.9",
    "@storybook/react": "^6.2.9",
    "@typescript-eslint/eslint-plugin": "^4.26.1",
    "@typescript-eslint/parser": "^4.26.1",
    "babel-eslint": "^10.1.0",
    "babel-loader": "^8.2.2",
    "chromatic": "^5.9.0",
    "eslint": "^7.27.0",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-import": "^2.23.4",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.4.0",
    "eslint-plugin-react": "^7.24.0",
    "eslint-plugin-react-hooks": "^4.2.0",
    "gatsby-plugin-netlify": "^3.10.0",
    "husky": "^6.0.0",
    "lint-staged": "^11.0.0",
    "prettier": "^2.3.1",
    "react-is": "^17.0.2",
    "typescript": "^4.3.2"
  },
```

```js
module.exports = {
    plugins: [
        "gatsby-plugin-react-helmet",
        {
            resolve: "gatsby-source-filesystem",
            options: {
                name: "images",
                path: `${__dirname}/src/images`,
            },
        },
        "gatsby-transformer-sharp",
        "gatsby-plugin-sharp",
        {
            resolve: "gatsby-plugin-manifest",
            options: {
                name: "oz-site",
                short_name: "oz",
                start_url: "/",
                background_color: "#fff",
                theme_color: "#fff",
                display: "minimal-ui",
                icon: "src/images/logo.svg",
            },
        },
        `gatsby-plugin-loadable-components-ssr`,
        {
            resolve: `gatsby-source-filesystem`,
            options: {
                name: `pages`,
                path: `${__dirname}/src/content`,
            },
        },
        {
            resolve: `gatsby-plugin-canonical-urls`,
            options: {
                siteUrl: `https://www.hazem.cool`,
                stripQueryString: true,
            },
        },
        {
            resolve: `gatsby-transformer-remark`,
        },
        {
            resolve: `gatsby-plugin-mdx`,
            options: {
                extensions: [".mdx", ".md"],
                gatsbyRemarkPlugins: [
                    {
                        resolve: `gatsby-remark-vscode`,
                        options: {},
                    },
                    {
                        resolve: `gatsby-remark-embedder`,
                        options: {},
                    },
                    {
                        resolve: `gatsby-remark-images`,
                        options: {
                            maxWidth: 736,
                            linkImagesToOriginal: false,
                            tracedSVG: {
                                color: "#2c5282",
                                background: "#bee3f8",
                            },
                        },
                    },
                    {
                        resolve: "gatsby-remark-audio",
                        options: {
                            preload: "auto",
                            loop: false,
                            controls: true,
                            muted: false,
                            autoplay: false,
                        },
                    },
                    `gatsby-remark-copy-linked-files`,
                    {
                        resolve: `gatsby-remark-autolink-headers`,
                    },
                ],
            },
        },
        `gatsby-plugin-image`,
        {
            resolve: "gatsby-plugin-react-svg",
            options: {
                rule: {
                    include: /images/,
                },
            },
        },
        `gatsby-plugin-twitter`,
        {
            resolve: `gatsby-plugin-advanced-sitemap`,
            query: `
            {
                allSitePage(filter: {context: {frontmatter: {draft: {ne: true}}}}) {
                    edges {
                        node {
                            context {
                                frontmatter {
                                    draft
                                }
                                slug
                            }
                            path
                        }
                    }
                }
            }
            `,
            mapping: {
                allSitePage: {
                    sitemap: "pages",
                },
            },
            exclude: [
                `/dev-404-page`,
                `/404`,
                `/404.html`,
                `/offline-plugin-app-shell-fallback`,
            ],
        },
        {
            resolve: `gatsby-plugin-robots-txt`,
            options: {
                policy: [{ userAgent: "*", allow: "/" }],
            },
        },
        {
            resolve: `gatsby-plugin-netlify`,
        },
        {
            resolve: "gatsby-plugin-offline",
            options: {
                navigateFallbackWhitelist: [/\/$/],
            },
        },
        `gatsby-plugin-remove-trailing-slashes`,
    ],
}
        
```

**Comment by graysonhicks** [comment]
Thanks @hazem3500 is the repo public?

**Comment by hazem3500** [comment]
Nope, unfortunately😅

**Comment by graysonhicks** [comment]
Okay, I am thinking this is a Typescript issue and [this solution may help](https://github.com/gregberge/loadable-components/issues/263#issuecomment-477996274). Let me know.

**Comment by hazem3500** [comment]
Cool 👍, thanks @graysonhicks XD, will test and tell you if it works!

**Comment by hazem3500** [comment]
Figured out the problem, I was importing `Loadable` with an uppercase instead of `loadable` 🤦‍♂️
```js
import Loadable from "@loadable/component"
```

instead of 

```js
import loadable from "@loadable/component"
```

**Comment by graysonhicks** [comment]
@hazem3500 Your second error may be fixed by 3.3.0

**Comment by hazem3500** [comment]
Just tried `3.3.0`, now I'm getting this error 😅

```
WebpackError: Cannot find module 'DIR\.cache\loadable-stats-build-javascript.json'
```

It's now missing from `.cache` instead of `public`

**Comment by graysonhicks** [comment]
@hazem3500 Are you on windows?  This doesn't seem related to the plugin?

**Comment by hazem3500** [comment]
Yes, I'm on Windows, the weird thing is that `loadable-stats-build-javascript.json` gets created in the root directory, is that supposed to happen?

**Comment by hazem3500** [comment]
I think this line is the culprit XD

https://github.com/graysonhicks/gatsby-plugin-loadable-components-ssr/blob/ec12fdbee2c8ae7e3fe8c4d935280ed00c8105c6/src/gatsby-node.js#L22

**Comment by graysonhicks** [comment]
Thanks @hazem3500 !  Can you describe your changes and the reasons why for long term clarity?

**Comment by graysonhicks** [comment]
Okay, makes sense, thanks!

**Comment by OPerel** [comment]
Hey
Any updates on this? Will the fixing PR be merged soon?
I'm having the same issue.
`loadable-stats-build-javascript.json` isn't created at all or is created at the root of the project.
I'm also on Windows, and using `gatsby: 3.13.0` and `gatsby-plugin-loadable-components-ssr: 3.3.0`

Thanks!

**Comment by hazem3500** [comment]
You can use [`patch-package`](https://www.npmjs.com/package/patch-package) until its merged I guess 😅

**Comment by graysonhicks** [comment]
@OPerel Yes, I will merge next week. 👍

**Comment by ref-kboone** [comment]
> @OPerel Yes, I will merge and release next week. 👍

Any idea when you can merge this? Next week we have a prod release and would like to have loadable components active without the patch approach.

**Comment by graysonhicks** [comment]
Merged and released with `3.4.0`.  Thanks all!
